core/modules/file/file.module

```diff
@@ -2777,101 +2777,68 @@ function file_get_type($file) {
 /**
  * Determines if a user may perform the given operation on the specified file.
  *
- * @param $op
+ * @param string $op
  *   The operation to be performed on the file. Possible values are:
+ *   - "create"
  *   - "view"
  *   - "download"
  *   - "update"
  *   - "delete"
- *   - "create"
- * @param $file
+ * @param object|string|NULL $file
  *   The file object on which the operation is to be performed, or file type
- *   (e.g. 'image') for "create" operation.
- * @param $account
+ *   (e.g. 'image').
+ * @param object $account
  *   Optional, a user object representing the user for whom the operation is to
  *   be performed. Determines access for a user other than the current user.
  *
- * @return
+ * @return bool
  *   TRUE if the operation may be performed, FALSE otherwise.
  */
 function file_access($op, $file = NULL, $account = NULL) {
-  $rights = &backdrop_static(__FUNCTION__, array());
-
-  if (!$file && !in_array($op, array('view', 'download', 'update', 'delete', 'create'), TRUE)) {
-    // If there was no file to check against, and the $op was not one of the
-    // supported ones, we return access denied.
+  // Filter out unsupported operations.
+  if (!in_array($op, array('create', 'view', 'download', 'update', 'delete'), TRUE)) {
     return FALSE;
   }
 
-  // If no user object is supplied, the access check is for the current user.
-  if (empty($account)) {
-    $account = $GLOBALS['user'];
-  }
-
-  // $file may be either an object or a file type. Since file types cannot be
-  // an integer, use either fid or type as the static cache id.
-  $cache_id = NULL;
-  if (!empty($file->id)) {
-    $cache_id = $file->id;
-  }
-  elseif (is_string($file) && $op == 'create') {
-    $cache_id = $file;
-  }
-  else {
-    $cache_id = backdrop_hash_base64(serialize($file));
-  }
-
-  // If we've already checked access for this file, user and op, return from
-  // cache.
-  if (isset($rights[$account->uid][$cache_id][$op])) {
-    return $rights[$account->uid][$cache_id][$op];
-  }
+  // If no $file is given or it's only the file type.
+  if (!$file || is_string($file)) {
```

Comment:
Actually, I'm still not sure if that makes sense, anyway (optional $file parameter handling). A big part of the permission checks needs an instance to check against.

So what's the use-case for this optional param and is it all worth it. I pretty much bloats this function, which is supposed to be only a wrapper for File::access. But File::access needs an instance - we can't do `NULL->access($op, $account)` :wink:

Reply:
If we return false if $file is a string then maybe don't accept it as a parameter in the docblock? Also it's checking it twice

Reply:
Strange ... didn't I mark this as resolved already? @herbdool  We decided to keep the optional param handling here.

The param can also be a string, when the operation is "create". Then it's the type. And entity_access (also in D7) provides this feature (check without entity or type), so removing would be a regression IMO.